swap/swap_test.go

```diff
@@ -123,86 +123,117 @@ func newTestBackend() *swapTestBackend {
 }
 
 // Test getting a peer's balance
-func TestPeerBalance(t *testing.T) {
+func TestBalance(t *testing.T) {
 	// create a test swap account
 	swap, testPeer, clean := newTestSwapAndPeer(t, ownerKey)
+	testPeerID := testPeer.ID()
 	defer clean()
 
-	// test for correct value
-	testPeer.setBalance(888)
-	b, err := swap.Balance(testPeer.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if b != 888 {
-		t.Fatalf("Expected peer's balance to be %d, but is %d", 888, b)
-	}
+	// test for correct balance
+	setBalance(t, testPeer, 888)
+	testBalance(t, swap, testPeerID, 888)
 
-	// test for inexistent node
-	id := adapters.RandomNodeConfig().ID
-	_, err = swap.Balance(id)
+	// test balance after change
+	setBalance(t, testPeer, 17000)
+	testBalance(t, swap, testPeerID, 17000)
+
+	// test balance for second peer
+	testPeer2 := addPeer(t, swap)
+	testPeer2ID := testPeer2.ID()
+
+	setBalance(t, testPeer2, 4)
+	testBalance(t, swap, testPeer2ID, 4)
+
+	// test balance for inexistent node
+	invalidPeerID := adapters.RandomNodeConfig().ID
+	_, err := swap.Balance(invalidPeerID)
 	if err == nil {
 		t.Fatal("Expected call to fail, but it didn't!")
 	}
 	if err != state.ErrNotFound {
 		t.Fatalf("Expected test to fail with %s, but is %s", "ErrorNotFound", err.Error())
 	}
 
-	// test for disconnected node
-	testPeer2 := newDummyPeer().Peer
-	swap.saveBalance(testPeer2.ID(), 333)
-	b, err = swap.Balance(testPeer2.ID())
-	if err != nil {
-		t.Fatal(err)
-	}
-	if b != 333 {
-		t.Fatalf("Expected peer's balance to be %d, but is %d", 333, b)
-	}
-}
+	// test balance for disconnected node
+	testPeer3 := newDummyPeer().Peer
+	testPeer3ID := testPeer3.ID()
+	err = swap.saveBalance(testPeer3ID, 777)
+	testBalance(t, swap, testPeer3ID, 777)
 
-// Test getting balances for all known peers
-func TestAllBalances(t *testing.T) {
-	// create a test swap account
-	swap, clean := newTestSwap(t, ownerKey, nil)
-	defer clean()
+	// test previous results are still correct
```

Comment:
What is the added value of these extra tests?

Reply:
the value isn't _huge_ (in fact, it's an arguably somewhat paranoic practice) but since we progressively deal with different peers, it's conceivable that after working with a new one, results from the previous peer could be affected (and therefore, be erroneous).

one could apply this logic for many more cases, though